### PR TITLE
Add 'comments' column to zvol/dataset

### PR DESCRIPTION
**Pull request by neharathod**
I have added a user-defined property 'volume_entity:comments' to zfs.
I have also added 'comments' field to the dataset/zvol forms.

### CODE REVIEW

File: gui/storage/views.py [hunk 1]
```diff
@@ -387,6 +387,7 @@ def dataset_create(request, fs):
             props['casesensitivity'] = cleaned_data.get(
                 'dataset_case_sensitivity'
             )
+	    dataset_comments = cleaned_data.get('dataset_comments')
```
Comment (williambr) [inline gui/storage/views.py]: Can you use spaces instead of tabs? That makes a big difference in python.

File: gui/storage/views.py [hunk 2]
```diff
@@ -411,6 +412,7 @@ def dataset_create(request, fs):
             errno, errmsg = notifier().create_zfs_dataset(
                 path=str(dataset_name),
                 props=props)
+	    notifier().zfs_set_option(name = str(dataset_name),item = "volume_entity:comments",value = dataset_comments)
```
Comment (williambr) [inline gui/storage/views.py]: Use spaces here too, also please make it PEP8 to conform with all the rest of the codebase, in this case means do not put spaces between equal signs in methods usage, and a space after comma.

e.g. name=str(dataset_name), item="volume...

File: gui/storage/views.py [hunk 3]
```diff
@@ -510,13 +512,15 @@ def zvol_create(request, parent):
             zvol_name = "%s/%s" % (parent, cleaned_data.get('zvol_name'))
             zvol_compression = cleaned_data.get('zvol_compression')
             props['compression'] = str(zvol_compression)
+	    zvol_comments = cleaned_data.get('zvol_comments')
```
Comment (williambr) [inline gui/storage/views.py]: Another tab

File: gui/storage/views.py [hunk 4]
```diff
@@ -510,13 +512,15 @@ def zvol_create(request, parent):
...
             if zvol_blocksize:
                 props['volblocksize'] = zvol_blocksize
             errno, errmsg = notifier().create_zfs_vol(
                 name=str(zvol_name),
                 size=str(zvol_size),
                 sparse=cleaned_data.get("zvol_sparse", False),
                 props=props)
+	    notifier().zfs_set_option(name = str(zvol_name),item = "volume_entity:comments",value = zvol_comments)
             if errno == 0:
```
Comment (williambr) [inline gui/storage/views.py]: Tab and PEP8

File: gui/storage/admin.py [hunk 5]
```diff
@@ -132,6 +132,13 @@ def get_datagrid_columns(self):
             'label': _('Status'),
             'sortable': False,
         })
+
+	columns.append({
```
Comment (williambr) [inline gui/storage/admin.py]: Tab

File: gui/api/resources.py [hunk 6]
```diff
@@ -909,6 +909,11 @@ def _get_children(self, bundle, vol, children, uid):
                     {},
                 ).get('compressratio', ('', '-'))[1]
 
+		data['comments'] = self.__zfsopts.get(
```
Comment (williambr) [inline gui/api/resources.py]: Use spaces instead of tabs.

**Comment by williambr** [comment]
Also, this pull request conflicts with master, please redo the pull request rebased on master.

Thanks

File: gui/storage/views.py (hunk 4 above)
Comment (splbio) [inline gui/storage/views.py]: Not sure if the FreeNAS project uses this, but several projects I've been involved in use flake8 with some subset of the options in place as a make target to be used before submitting code.  We also use Travis to cause an error if someone submits code that flake8 disagrees with.  Just need to carefully pick which flake8 tests you want to run, some are really overly annoying.

**Comment by jceel** [comment]
Name of ZFS property to store comments in should be `org.freenas:description` to be forward-compatible with FreeNAS 10.

**Comment by neharathod** [comment]
Okay, thank you for your feedback.
I'll create a new pull request against the master branch conforming with the PEP8 standards.
Also, as per the convention for ZFS user-defined property of module:property, I have named the new comments property as storage:comments

**Comment by jceel** [comment]
`storage:comments` is wong. you shall use `org.freenas:description`.

**Comment by neharathod** [comment]
Okay sure!

**Pull request closed without merging**